src/wallet/wallet.cpp

```diff
@@ -2080,8 +2082,15 @@ void CWallet::AvailableCoins(vector<COutput>& vCoins, bool fOnlyConfirmed, const
             }
 
             for (unsigned int i = 0; i < pcoin->tx->vout.size(); i++) {
-                isminetype mine = IsMine(pcoin->tx->vout[i]);
-                if (!(IsSpent(wtxid, i)) && mine != ISMINE_NO &&
+                isminetype mine = ISMINE_NO;
+                std::map<CScript, isminetype>::const_iterator im = mapOutputIsMine.find(pcoin->tx->vout[i].scriptPubKey);
+                if (im != mapOutputIsMine.end()) {
+                    mine = im->second;
+                } else {
+                    mine = mapOutputIsMine[pcoin->tx->vout[i].scriptPubKey] = IsMine(pcoin->tx->vout[i]);
+                }
+
+                if (mine != ISMINE_NO && !(IsSpent(wtxid, i)) &&
```

Comment:
This would seem like a good idea, especially if the continues were documented with new comments. Not really on-topic for this PR, however.